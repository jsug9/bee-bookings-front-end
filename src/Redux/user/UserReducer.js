/* eslint-disable no-param-reassign */
import axios from 'axios';
import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

const signInEndpoint = 'https://bee-store.herokuapp.com/api/v1/signin';
const signUpEndpoint = 'https://bee-store.herokuapp.com/api/v1/signup';

export const signIn = createAsyncThunk('user/signIn', async (username) => {
  const userParams = { user: { username } };
  const { data } = await axios.post(signInEndpoint, userParams);
  return data;
});

export const sum = () => 'from user reducer';

export const signUp = createAsyncThunk('user/signUp', async (username) => {
  const userParams = { user: { username } };
  const { data } = await axios.post(signUpEndpoint, userParams);
  return data;
});

const initialState = {
  username: null,
  userId: null,
  isLoading: false,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    logOutUser: (state) => {
      state.username = null;
      state.userId = null;
      state.createdUser = null;
    },
  },
  extraReducers: {
    [signIn.pending]: (state) => {
      state.isLoading = true;
    },
    [signIn.fulfilled]: (state, action) => {
      state.username = action.payload.username;
      state.userId = action.payload.user_id;
      state.isLoading = false;
    },
    [signIn.rejected]: (state) => {
      state.isLoading = false;
    },
    [signUp.pending]: (state) => {
      state.isLoading = true;
    },
    [signUp.fulfilled]: (state, action) => {
      state.username = action.payload.username;
      state.userId = action.payload.user_id;
      state.isLoading = false;
      state.createdUser = true;
    },
    [signUp.rejected]: (state) => {
      state.isLoading = false;
    },
  },
});

export const { logOutUser } = userSlice.actions;
export default userSlice.reducer;
