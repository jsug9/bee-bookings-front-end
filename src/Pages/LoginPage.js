import '../Styles/login_page.scss';
import { TextField } from '@mui/material';
import { useDispatch, useSelector } from 'react-redux';
import React, { useState } from 'react';
import { toggleSignup } from './SignupPage';
import { signIn } from '../Redux/user/UserReducer';

// Toggles visibility of the login page overlay
// NO TOUCHY: specially Augusto
export const toggleLogin = () => {
  const loginPage = document.getElementById('loginPage');
  loginPage.classList.toggle('invisible');
  loginPage.classList.toggle('flex');
};

// Toggles visibility of the login page overlay when the user clicks the sign up link
// NO TOUCHY: specially Augusto
const signupToggle = () => {
  toggleLogin();
  toggleSignup();
};

const LoginPage = () => {
  const [username, setUsername] = useState('');
  const user = useSelector((state) => state.user);

  const dispatch = useDispatch();
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(signIn(username));
  };
  return (
    <div id="loginPage" className={user.userId ? 'hide' : 'invisible'}>
      <div id="loginClose" role="presentation" onClick={toggleLogin}>X</div>
      <div className="loginForm mobile">
        <h1>Buzz In</h1>
        <section className="loginSection" id="loginSection">
          <form className="loginForm" method="post" onSubmit={handleSubmit}>
            <TextField
              style={{ backgroundColor: 'white', borderRadius: '4px' }}
              type="text"
              name="username"
              id="username"
              label="Enter your username"
              required
              maxLength="30"
              value={username}
              onChange={(e) => setUsername(e.target.value)}
            />
            <span className="error" aria-live="polite" />
            <button type="submit" id="submitLogin" className="loginBtn">Let&apos;s Go!</button>
          </form>
          <div className="signupLink">
            <p id="signup" role="presentation" onClick={signupToggle}>Sign Up</p>
          </div>
        </section>
      </div>
    </div>
  );
};

export default LoginPage;
