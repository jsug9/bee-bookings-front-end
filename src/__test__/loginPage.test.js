import LoginPage from '../Pages/LoginPage';
import renderWithProviders, { screen } from './test-utils';
import { signIn } from '../Redux/user/UserReducer';
import server from '../mswMocks/server';
import realStore from '../Redux/configureStore';

// Establish API mocking before all tests.
beforeAll(() => server.listen());
// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers());
// Clean up after the tests are finished.
afterAll(() => server.close());

it('login page renders to the page', () => {
  const tree = renderWithProviders(
    <LoginPage />,
  );
  expect(tree).toMatchSnapshot();
});

it('Checks that the login Page displays to the screen', () => {
  renderWithProviders(
    <LoginPage />,
  );
  expect(screen.getByText(/Buzz In/)).toBeInTheDocument();
});

it('Tests the sign in functionality of the redux store', async () => {
  const store = realStore;
  await store.dispatch(signIn('AaronIsCool'));

  renderWithProviders(
    <LoginPage />, { store },
  );

  expect(store.getState().user.username).toStrictEqual('AaronIsCool');
});
