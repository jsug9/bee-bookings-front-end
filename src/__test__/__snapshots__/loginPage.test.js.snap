// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page renders to the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="invisible"
        id="loginPage"
      >
        <div
          id="loginClose"
          role="presentation"
        >
          X
        </div>
        <div
          class="loginForm mobile"
        >
          <h1>
            Buzz In
          </h1>
          <section
            class="loginSection"
            id="loginSection"
          >
            <form
              class="loginForm"
              method="post"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                maxlength="30"
                style="background-color: white; border-radius: 4px;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="username"
                  id="username-label"
                >
                  Enter your username
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="username"
                    name="username"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Enter your username
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <span
                aria-live="polite"
                class="error"
              />
              <button
                class="loginBtn"
                id="submitLogin"
                type="submit"
              >
                Let's Go!
              </button>
            </form>
            <div
              class="signupLink"
            >
              <p
                id="signup"
                role="presentation"
              >
                Sign Up
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="invisible"
      id="loginPage"
    >
      <div
        id="loginClose"
        role="presentation"
      >
        X
      </div>
      <div
        class="loginForm mobile"
      >
        <h1>
          Buzz In
        </h1>
        <section
          class="loginSection"
          id="loginSection"
        >
          <form
            class="loginForm"
            method="post"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              maxlength="30"
              style="background-color: white; border-radius: 4px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="username"
                id="username-label"
              >
                Enter your username
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="username"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Enter your username
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <span
              aria-live="polite"
              class="error"
            />
            <button
              class="loginBtn"
              id="submitLogin"
              type="submit"
            >
              Let's Go!
            </button>
          </form>
          <div
            class="signupLink"
          >
            <p
              id="signup"
              role="presentation"
            >
              Sign Up
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
